doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,84 @@
+# Architecture
+
+This chapter describes how Nix works.
+It should help users understand why Nix behaves as it does, and it should help developers understand how to modify Nix and how to write similar tools.
+
+## Overview
+
+Nix consists of [hierarchical layers].
+
+[hierarchical layers]: https://en.m.wikipedia.org/wiki/Multitier_architecture#Layers
+
+```
++---------------------------------------------------------------+
+| Nix             .-------------------------.                   |
+|                 | commmand line interface |------.            |
+|                 '-------------------------'      |            |
+|                              |                   |            |
+|                          evaluates               |            |
+|                              |                manages         |
+|                              V                   |            |
+|                 .-------------------------.      |            |
+|                 | configuration language  |      |            |
+|                 '-------------------------'      |            |
+|                              |                   |            |
+|                         evaluates to             |            |
+|                              |                   V            |
+| +----------------------------|------------------------------+ |
+| | store                      |                              | |
+| |            referenced by   V       builds                 | |
+| | .-------------.      .------------.      .--------------. | |
+| | | build input |----->| build plan | ---->| build result | | |
+| | '-------------'      '------------'      '--------------' | |
+| +-----------------------------------------------------------+ |
++---------------------------------------------------------------+
```

Comment:
It would be good if this picture includes the names of the components: `libstore`, `libexpr`, `libcmd / src/nix`, ...